### PR TITLE
Updates variable costs

### LINKED ISSUE #14: Add variable costs for fuels
Add variable costs for the model in a new jupyter notebook.  Base costs on World Bank Commodity Outlook.

Fuel production technology names will need to come from the powerplant_data notebook or from the output.csv.

Include small variable cost for international trading hubs for the commodities.

### PR DESCRIPTION
**Pull request by trevorb1**
### Description

In this PR I have: 
- Refactored the variable costs module to make custom updates a little easier 
- Corrected international variable costs data

The logic for international variable costs data is: 
- All international variable costs (mining techs the operate on mode 2) are assigned a standard international cost factor 
- If a country level fuel resource limit exists, this cost factor remains 
- If a country level fuel resource limit does not exist, the country can NOT contribute to international markets (via a very high variable cost to operate on mode 2) 

### Issue Ticket Number

Closes #14 

### Documentation

na

### CODE REVIEW

File: workflow/scripts/osemosys_global/powerplant/fuel_limits.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        s.name = "VALUE"
+        return s
+
+    limits = fuel_limits.copy()
+    limits = limits[limits.VALUE > 0.001].copy()  # assume these are zero
+
+    # not sure what the year column represents?
+    limits = limits.drop(columns="YEAR").drop_duplicates(
+        subset=["FUEL", "COUNTRY"], keep="last"
+    )
```
Comment (trevorb1) [inline workflow/scripts/osemosys_global/powerplant/fuel_limits.py:59]: Here is where I drop the years in fuel limits

**Comment by maartenbrinkerink** [comment]
@trevorb1 The workflow runs fine on my end. Regarding the multi-year fuel limits I think we should confirm this with @abhishek0208 today. If constant values are the way to go, and not by year, than I think we need to change this in `fuel_limits.csv` as it gives the wrong impression to users that fuel limits can be changed by year (yet filtered out within the workflow). 

Single rules for var costs/fuel limits is grand as long as they stay within the powerplant folder (which they are).

**Comment by trevorb1** [comment]
@maartenbrinkerink okay, great! I believe this has been updated to reflect our discussion earlier today. Specifically: 
- The model period limit constraint has been removed 
- A yearly activity limit on each of the domestic mining technologies has been applied, following the values from the `fuel_limits.csv` file. The same restriction applies though; if no user limits are applied, the max activity is set to zero and the country must import from the international market

**Comment by maartenbrinkerink** [comment]
Good stuff @trevorb1. I'll merge this into the target updates PR and solve any potential conflicts before merging it into master.

**Pull request merged**